RDKShell/RDKShell.cpp

```diff
@@ -5740,6 +5759,101 @@ namespace WPEFramework {
             returnResponse(status);
         }
 
+        uint32_t RDKShell::checkpointWrapper(const JsonObject& parameters, JsonObject& response)
+        {
+            LOGINFOMETHOD();
+            bool status = false;
+            if (parameters.HasLabel("callsign"))
+            {
+                std::string callsign = parameters["callsign"].String();
+                bool isApplicationBeingDestroyed = false;
+
+                gLaunchDestroyMutex.lock();
+                if (gDestroyApplications.find(callsign) != gDestroyApplications.end())
+                {
+                    isApplicationBeingDestroyed = true;
+                }
+                if (gExternalDestroyApplications.find(callsign) != gExternalDestroyApplications.end())
+                {
+                    isApplicationBeingDestroyed = true;
+                }
+                gLaunchDestroyMutex.unlock();
+
+                if (isApplicationBeingDestroyed)
+                {
+                    std::cout << "ignoring checkpoint for " << callsign << " as it is being destroyed " << std::endl;
+                    status = false;
+                    response["message"] = "failed to checkpoint application, is being destroyed";
+                    returnResponse(status);
+                }
+
+                std::thread requestsThread =
+                std::thread([=]()
+                {
+                    auto systemServiceConnection = RDKShell::getThunderControllerClient();
```

Comment:
systemServiceConnection is misleading,  please rename this

Reply:
changed to thunderController